internal/services/communication/communication_service_email_domain_association_resource_test.go

```diff
@@ -0,0 +1,213 @@
+			),
+			ExpectNonEmptyPlan: true,
+		},
+	})
+}
+
+func (r CommunicationServiceEmailDomainAssociationResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	exists := false
+
+	id, err := commonids.ParseCompositeResourceID(state.ID, &communicationservices.CommunicationServiceId{}, &domains.DomainId{})
+	if err != nil {
+		return &exists, fmt.Errorf("parsing ID: %w", err)
+	}
+
+	serviceClient := client.Communication.ServiceClient
+	existingCommunicationService, err := serviceClient.Get(ctx, *id.First)
+	if err != nil && !response.WasNotFound(existingCommunicationService.HttpResponse) {
+		return &exists, fmt.Errorf("checking for the presence of existing PrivateEndpoint %q: %+v", id.First, err)
+	}
+
+	if response.WasNotFound(existingCommunicationService.HttpResponse) {
+		return &exists, fmt.Errorf("PrivateEndpoint %q does not exsits", id.First)
+	}
+
+	input := existingCommunicationService
+	if input.Model != nil && input.Model.Properties != nil && input.Model.Properties.LinkedDomains != nil {
+		exists = slices.Contains(*input.Model.Properties.LinkedDomains, id.Second.ID())
+	}
+
+	return &exists, nil
+}
+
+func (r CommunicationServiceEmailDomainAssociationResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_communication_service_email_domain_association" "test" {
+  communication_service_id = azurerm_communication_service.test.id
+  email_service_domain_id  = azurerm_email_communication_service_domain.test.id
+}
+`, r.template(data))
+}
+
+func (r CommunicationServiceEmailDomainAssociationResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-communicationservice-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_communication_service" "test" {
+  name                = "acctest-CommunicationService-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  data_location       = "United States"
+
+  tags = {
+    env = "Test2"
+  }
+}
+
+resource "azurerm_email_communication_service" "test" {
+  name                = "acctest-CommunicationService-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  data_location       = "United States"
+}
+
+resource "azurerm_email_communication_service_domain" "test" {
+  name             = "AzureManagedDomain"
+  email_service_id = azurerm_email_communication_service.test.id
+
+  domain_management = "AzureManaged"
+}
+
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r CommunicationServiceEmailDomainAssociationResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_communication_service_email_domain_association" "import" {
+  communication_service_id = azurerm_communication_service.test.id
+  email_service_domain_id  = azurerm_email_communication_service_domain.test.id
```

Comment:
These values should be references to the original resource values:
```suggestion
  communication_service_id = azurerm_communication_service_email_domain_association.test.communication_service_id
  email_service_domain_id  = azurerm_communication_service_email_domain_association.test.email_service_domain_id
```